### PR TITLE
docs: add article covering asset signature verification

### PR DESCRIPTION
**Pull request by mohammed90**

### CODE REVIEW

File: src/docs/markdown/install.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ Our [official packages](https://github.com/caddyserver/dist) come only with the
 
 1. Obtain a Caddy binary:
 	- [from releases on GitHub](https://github.com/caddyserver/caddy/releases) (expand "Assets")
+		- Refer to [Verifying Asset Signatures](/docs/signature-verification) for how to verify the asset signature
```
Comment (mohammed90) [inline src/docs/markdown/install.md:40]: I'm not exactly sure what's the best way to link to the article from here, so I put up a sub-sub-bullet for now 🤷🏼‍♂️

File: src/docs/markdown/signature-verification.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
+---
+title: Verifying Asset Signatures
+---
+
+# Signature Verification
+
+CI/CD release artifacts are now signed using project Sigstore technology, which issues certificates containing details about the subject to whom the certificate is issues amongst others. You can start by inspecting the certificate used to sign your artifact of choice. The certificates are base64 encoded, so you first have to base64 decode it to receive the pem file. In the example, we'll work with the `caddy_2.6.0_checksums.txt` artifact and assume *nix environemnt.
```
Comment (mholt) [inline src/docs/markdown/signature-verification.md]:
```suggestion
CI/CD release artifacts are now signed using project [Sigstore](https://www.sigstore.dev/) technology, which issues certificates containing details about the subject to whom the certificate is issued. You can start by inspecting the certificate used to sign your artifact of choice. The certificates are base64-encoded, so you first have to base64-decode it to receive the PEM file. In this example, we'll work with the `caddy_2.6.0_checksums.txt` artifact and assume a Linux-like environment.
```

File: src/docs/markdown/signature-verification.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  "LogIndex": 3618623,
+  "IntegratedTime": 1663694226,
+  "UUID": "04deb84e5a73ba75ea69092c6d700eaeb869c29cae3e0cf98dbfef871361ed09",
+  "LogID": "c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"
+}
+```
+
+Notice how the value of `.Body.HashedRekordObj.signature.content` matches the content of the signature generated in our CI and available in the file `caddy_2.6.0_checksums.txt.sig`. Moreover, the certificate used and downloaded is also stored in the Rekor server and available in the response at `.Body.HashedRekordObj.signature.publicKey.content` and matches the string we have in the file `caddy_2.6.0_checksums.txt.pem`. We can take one step further and check how `.Body.HashedRekordObj.data.hash.value` matches the output of the command `sha256sum ./caddy_2.6.0_checksums.txt`. So by now we have matching certs, matching signatures, and matching checksums (of the file containing the checksums of the archives but not of itself; this checksum is provided and recorded externally via Sigstore ecosystem). All of this is publicly recorded in transparency logs for the general public to validate.
+
+## What If...
```
Comment (mholt) [inline src/docs/markdown/signature-verification.md]: Can we put a more useful header here? :)

File: src/docs/markdown/signature-verification.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+<pre><code class="cmd"><span class="bash">rekor-cli search --artifact ./caddy_2.6.0_checksums.txt --format json | jq -r '.UUIDs[0]'</span>
+Found matching entries (listed by UUID):
+362f8ecba72f432604deb84e5a73ba75ea69092c6d700eaeb869c29cae3e0cf98dbfef871361ed09</code></pre>
+
+Note how the UUID matches the one encountered in the earlier section for the same file. Like we did in the earlier section, we can query Rekor for the entry details of this UUID:
+
+<pre><code class="cmd bash">rekor-cli get --uuid 04deb84e5a73ba75ea69092c6d700eaeb869c29cae3e0cf98dbfef871361ed09 --format json | jq -r '.'</code></pre>
+
+However, we can shortcircuit the lookup by running this line to merge the two separate commands into a one-liner:
```
Comment (mholt) [inline src/docs/markdown/signature-verification.md]:
```suggestion
However, we can short-circuit the lookup by running this line to merge the two separate commands into a one-liner:
```

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks so much for writing this -- very useful info and I never would have figured this out on my own! I'm sure many of our users will appreciate this.

How would you feel about adding a brief section to the beginning that explains the value of what we're doing? It shouldn't be too long or detailed. But basically introduce readers to why signatures may be important to them.

Could we add a section maybe at the end that explains in summary what to look for? And what it could mean if there's a mismatch? And maybe even some recommendations of what to do?

File: src/docs/markdown/signature-verification.md (hunk 3 above)
Comment (mohammed90) [reply src/docs/markdown/signature-verification.md]: I've changed the header and added more details to this section

**Review by mholt — APPROVED** [review]
Awesome, Mohammed, this is awesome. Thank you for writing this up and contributing yet again to the docs!

**Pull request merged**